File: schemas/podcast-platform/convex/funding_links.ts
// funding_links: external funding or support URLs associated with a show.
// See README.md for full design rationale.

import { defineTable } from "convex/server";
import { v } from "convex/values";

export const funding_links = defineTable({
  showId: v.id("shows"),
  url: v.string(),
  title: v.string(),
  position: v.number(),
}).index("by_show_id_position", ["showId", "position"]);

File: schemas/podcast-platform/spacetimedb/funding_links.rs
// funding_links: External funding and donation links associated with a show.
// See README.md for full design rationale.

use spacetimedb::Timestamp;

#[spacetimedb::table(name = funding_links, public)]
pub struct FundingLink {
    #[primary_key]
    pub id: String, // UUID
    #[index(btree)]
    pub show_id: String, // UUID — FK → shows.id (cascade delete)
    pub url: String,
    pub title: String,
    pub position: i32,
    pub created_at: Timestamp,
}
// Composite index: (show_id, position) — not supported, enforce in reducer logic
